servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/PipelinedStreamingHttpConnection.java

```diff
@@ -40,16 +41,40 @@
     @Override
     protected Publisher<Object> writeAndRead(Publisher<Object> requestStream,
                                              @Nullable final FlushStrategy flushStrategy) {
-        if (flushStrategy == null) {
-            return connection.write(requestStream);
-        } else {
-            // TODO(scott): if we can remove the flush state on the connection we can simplify the control flow here.
-            return Publisher.defer(() -> {
+        assert connectionContext().protocol().major() <= 1 : "Unexpected protocol version";
+        return (flushStrategy == null ? connection.write(requestStream) :
+            Publisher.defer(() -> {
                 final Cancellable resetFlushStrategy = connection.updateFlushStrategy(
```

Comment:
`// TODO(scott): if we can remove the flush state on the connection we can simplify the control flow here.` -> not necessary to retain but we will be reworking this in the next few months so I assume this is understood :)